crates/solvers/src/domain/naive.rs

```diff
@@ -0,0 +1,86 @@
+//! "Naive" solver implementation.
+//!
+//! The naive solver is a solver that collects all orders over a single token
+//! pair, and matches the excess over a Uniswap V2 pool.
+
+use {
+    super::Solver,
+    crate::{
+        boundary,
+        domain::{auction, liquidity, order, solution},
+    },
+    futures::future::{BoxFuture, FutureExt},
+    std::collections::HashMap,
+};
+
+pub struct Naive;
+
+impl Solver for Naive {
+    /// Solves the specified auction, returning a vector of all possible
+    /// solutions.
+    fn solve(&self, auction: auction::Auction) -> BoxFuture<Vec<solution::Solution>> {
+        let groups = group_by_token_pair(&auction);
+        let solutions = groups
+            .values()
+            .filter_map(|group| boundary::naive::solve(&group.orders, group.liquidity))
+            .collect();
+
+        async { solutions }.boxed()
+    }
+}
+
+#[derive(Debug)]
+struct Group<'a> {
+    orders: Vec<&'a order::Order>,
+    liquidity: &'a liquidity::Liquidity,
+    pool: &'a liquidity::constant_product::Pool,
+}
+
+type Groups<'a> = HashMap<liquidity::TokenPair, Group<'a>>;
+
+/// Groups an auction by token pairs, where each group contains all orders over
+/// the token pair as well as the **deepest** constant product pool.
```

Comment:
It's not clear to me what "deepest" means in this context. From reading the implementation presumably it means that it has the highest `k`? Do you think there may be a more explicit term for this?

Reply:
12e2d3205a6b8ac29ac655129fd3c4d687e74ef5